import "common.proto";

//1832
message GetActGiftRequest{
    required int32 act_id = 1;
	optional int32 quantity = 2;
}

message GetActGiftResponse{
    required CommonResponse res = 1;
    optional GameResourcesResponse gain = 2;
    optional GameResourcesResponse consume = 3;
}

//1831
message GetActInfoRequese{
    required int32 act_type = 1;
}

message GetActInfoResponse{
    required CommonResponse res = 1;
    repeated int32 received_act_ids = 2;
    optional int32 times = 3;
    optional int32 act_type = 4;
}

message FundActivityInfo{
	optional int32 act_id = 1;
	optional int32 state = 2;
	optional int32 accumulate_days = 3;
	optional int32 recharge = 4;
	optional int32 max_single_recharge = 5;
}

message GetFundActivityResponse{
	repeated FundActivityInfo info = 1;
}

// 2501 公会活动初始化信息
message GuildActivityInitResponse{
	repeated GuildActivityInfo acts = 1;
}

message GuildActivityInfo{
	required int32 act_id = 1; //活动id
	optional int32 act_times = 2; //活动次数
	optional bool finished = 3; //是否已经领取
}

// 2502 获取活动奖励
message GuildActivityGetRewardRequest{
	required int32 act_id = 1; //活动id
}
message GuildActivityGetRewardResponse{
	required CommonResponse res = 1; 
    optional GameResourcesResponse gain = 2; //活动奖励
}
